### PR TITLE
Add documentation for helm installation mode

### PR DESCRIPTION
**Pull request by michi-covalent**
Ref: https://github.com/cilium/cilium-cli/issues/1396

### CODE REVIEW

**Comment by michi-covalent** [comment]
@tklauser @asauber let's use this pull request to discuss the plan for deprecating the classic installation mode.

File: README.md [hunk 1]
```diff
@@ -336,3 +336,60 @@ Install a Cilium in a cluster and enable encryption with IPsec
...
+- [x] `config`
+- [x] `connectivity`
+- [x] `context`
+- [x] `help`
+- [x] `hubble`
+- [x] `install`
+- [x] `status`
+- [x] `sysdump`
+- [x] `uninstall`
+- [ ] `upgrade`
```
Comment (tklauser) [inline README.md:394]: I think this is supported now with https://github.com/cilium/cilium-cli/pull/1492 in place.

File: README.md [hunk 2]
```diff
@@ -336,3 +336,60 @@ Install a Cilium in a cluster and enable encryption with IPsec
...
+related to Cilium. In the new `helm` mode, `cilium-cli` delegates all the installation state
+management to Helm. This enables you to use `cilium-cli` and `helm` interchangeably to manage your
+Cilium installation, while taking advantage of `cilium-cli`'s advanced features such as Cilium
+configuration auto-detection.
+
+Our current plan is to:
+
+- v0.14: Introduce the `helm` installation mode as an opt-in feature.
+- v0.15: Enable the `helm` installation mode by default, and deprecate the `classic` installation mode.
+- v1.0: Remove the `classic` installation mode.
```
Comment (tklauser) [inline README.md:353]: The plan SGTM. I wonder whether we should also mention or link this in the *Releases* section above.

File: README.md (hunk 1 above)
Comment (michi-covalent) [reply README.md]: done ✅

File: README.md (hunk 2 above)
Comment (michi-covalent) [reply README.md:357]: done ✅ added a link to this section in the `Release` section.

**Review by tklauser — APPROVED** [review]

**Comment by michi-covalent** [comment]
the button is green.

**Pull request merged**